### DISCUSSION TITLE
Can I dynamically set the kc_idp_hint parameter for a client?

### DISCUSSION THREAD

**Discussion by oculos** [keycloak/keycloak#29587] [Q&A]
Hi,

I wonder if I could use an attribute, client or realm name, to send as kc_idp_hint. Or could set something on the authentication flow to end a custom kc_idp_hint?

My goal here is not to set the kc_idp_hint on my client configuration, but to choose one dynamically based by which client the user is connecting from. 
Let's say the client "a" has an attribute "b". I'd like then to send "kc_idp_hint=b" when a user from that client clicks on an idp.

**Comment by danielFesenmeyer** [comment]
I don't think Keycloak does support this out of the box.
Maybe you can get some inspiration from this extension provided by @sventorben: https://github.com/sventorben/keycloak-home-idp-discovery

**Reply by oculos** [reply]
Very nice tip @danielFesenmeyer. I’ll look Into it! Thanks!
